Flames/flames_drv.py

```diff
@@ -293,78 +295,125 @@ def checkSequence(self, firingSequence):
+            ser.close()
+            ser = None
+            print("Error sending bangCommandSequence to poofer controller boards", e)
+            traceback.print_exc(file=sys.stdout)
+
+    def disablePoofer(msgObj):
+        self.disabled_poofers.add(msgObj["name"])
+        event_manager.postEvent({"msgType":"poofer_disabled", "id":msgObj["name"]})
+
+    def enablePoofer(msgObj):
+        enabled_poofer = self.disabled_poofers.pop(msgObj["name"])
+        if enabled_poofer != None:
+            event_manager.postEvent({"msgType":"poofer_enabled", "id":msgObj["name"]})
+
+    def resumeAll():
+        self.isStopped = False
+        event_manager.postEvent({"msgType":"global_resume", "id":"all?"})
+
+    def stopAll():
+        self.isStopped = True
+        # TODO: send bang command to stop all poofers
+        self.pooferEvents = list() # reset all pooferEvents
+        event_manager.postEvent({"msgType":"global_pause", "id":"all?"})
+
+    def startFlameEffect(msgObj):
+        if self.checkSequence(msgObj):
+            setUpEvent(msgObj)
+            event_manager.postEvent({"msgType":"sequence_start", "id":msgObj["name"]})
+
+    def stopFlameEffect(msgObj):
+        #  TODO: Need to figure out how to stop this,
+        # since now this is also removing commands to stop poofers
+        event_manager.postEvent({"msgType":"sequence_stop", "id":msgObj["name"]})
+        filter(lambda p: p.sequence != msgObj["name"], pooferEvents)
+
+    def setUpEvent(msgObj):
+        # Takes a sequence object, and add to self.pooferEvents the bang commands
+        # to turn on and to turn off the specified poofers.
+        # The obect added to self.pooferEvents is of format:
+        # # { "sequence":"sequenceName", "time":"1502068215.5",
+        # "bangCommandList":["!0011~21.", "!0021~21."] }
+
+        sequenceName = msgObj["name"]
+        sequence = self.pattern_manager.getPattern(sequenceName)
+
+        duration = sequence["duration"]
+        events = sequence["events"]
+        firstFiringTime = time.time()
+
+        if not self.isStopped:
+            for event in events:
+                ids = events["ids"]
+                startTime = firstFiringTime + event["startTime"]
+                endTime = startTime + event["duration"]
+
+                addresses = [pooferMapping[a] for a in ids].sort()
+                bangCommandList = makeBangCommandList(addresses)
+
+                pooferEvent = {}
+                pooferEvent["sequence"] = sequenceName
+                pooferEvent["time"] = startTime
+                pooferEvent["bangCommandList"] = bangCommandList["on"]
+
+                endPooferEvent = {}
+                endPooferEventendPooferEvent["sequence"] = sequenceName
+                endPooferEvent["time"] = endTime
+                endPooferEvent["bangCommandList"] = bangCommandList["off"]
+
+                # TODO: need to figure out best way to sort this thing
+                self.pooferEvents.append(pooferEvent)
```

Comment:
list.sort() is your friend here. It takes a sort parameter, which you can set as the item/attribute getter for the time. Or you can do something more complex.